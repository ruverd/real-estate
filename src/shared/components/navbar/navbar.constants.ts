export const NAVBAR_LINKS = [
  {
    label: "Home",
    href: "/",
  },
  {
    label: "About us",
    href: "/about",
  },
  {
    label: "Contact us",
    href: "/contact",
  },
];

export const NAVBAR_CONTACT_INFO = {
  phone: "(257) 388-6895",
};
