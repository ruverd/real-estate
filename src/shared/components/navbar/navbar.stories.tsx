import type { Meta, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";

import { Navbar } from "./navbar";

const meta: Meta<typeof Navbar> = {
  title: "Components/Navbar",
  component: Navbar,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Navbar>;

export const Default: Story = {
  render: () => <Navbar />,
}; 