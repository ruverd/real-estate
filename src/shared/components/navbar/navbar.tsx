import { Phone } from "lucide-react";
import { Link } from "react-router-dom";

import { Logo } from "../logo/logo";
import { NavigationMenu, NavigationMenuItem, NavigationMenuLink, NavigationMenuList, navigationMenuTriggerStyle } from "../ui/navigation-menu";
import { NAVBAR_CONTACT_INFO, NAVBAR_LINKS } from "./navbar.constants";

export const Navbar = () => {
  return (
    <div className="container px-4 md:px-0 h-16 mx-auto flex items-center justify-between">
      <Logo />
      <NavigationMenu>
        <NavigationMenuList>
          {NAVBAR_LINKS.map((link) => (
            <NavigationMenuItem key={link.href}>
              <Link to={link.href}>
                <NavigationMenuLink className={navigationMenuTriggerStyle()}>
                  {link.label}
                </NavigationMenuLink>
              </Link>
            </NavigationMenuItem>
          ))}
        </NavigationMenuList>
      </NavigationMenu>

      <div className="items-center gap-2 hidden md:flex">
        <Phone className="size-4" />
        <span className="font-sans font-medium text-foreground">{NAVBAR_CONTACT_INFO.phone}</span>
      </div>
    </div>
  );
};
