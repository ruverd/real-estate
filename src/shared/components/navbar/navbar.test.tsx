import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { vi } from "vitest";

import { Navbar } from "./navbar";
import { NAVBAR_CONTACT_INFO, NAVBAR_LINKS } from "./navbar.constants";

vi.mock("../logo/logo", () => ({
  Logo: () => <div data-testid="logo-component">Logo</div>,
}));

const renderNavbar = () => {
  return render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
};

describe("Navbar", () => {
  it("should render the component correctly", () => {
    renderNavbar();
    
    const navbarContainer = screen.getByRole("navigation").closest("div");

    expect(navbarContainer).toBeInTheDocument();
  });

  it("should render the Logo component", () => {
    renderNavbar();
    
    const logoComponent = screen.getByTestId("logo-component");

    expect(logoComponent).toBeInTheDocument();
  });

  it("should render all navigation links correctly", () => {
    renderNavbar();
    
    NAVBAR_LINKS.forEach((link) => {
      const linkElement = screen.getByText(link.label);

      expect(linkElement).toBeInTheDocument();
    });
  });

  it("should render contact information", () => {
    renderNavbar();
    
    const phoneNumber = screen.getByText(NAVBAR_CONTACT_INFO.phone);

    expect(phoneNumber).toBeInTheDocument();
    
    const phoneIcon = phoneNumber.parentElement?.querySelector("svg");

    expect(phoneIcon).toBeInTheDocument();
  });

  it("should hide contact information on mobile screens", () => {
    renderNavbar();
    
    const contactContainer = screen.getByText(NAVBAR_CONTACT_INFO.phone).closest("div");

    expect(contactContainer).toHaveClass("hidden md:flex");
  });

  it("should ensure navigation links are accessible", () => {
    renderNavbar();
    
    const navigationMenu = screen.getByRole("navigation");
    const navigationItems = screen.getAllByRole("listitem");

    expect(navigationMenu).toBeInTheDocument();
    expect(navigationItems.length).toBe(NAVBAR_LINKS.length);
  });

  it("should match snapshot", () => {
    const { container } = renderNavbar();

    expect(container.firstChild).toMatchSnapshot();
  });
});
