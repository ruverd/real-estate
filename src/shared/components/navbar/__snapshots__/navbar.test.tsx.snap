// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should match snapshot 1`] = `
<div
  class="container px-4 md:px-0 h-16 mx-auto flex items-center justify-between"
>
  <div
    data-testid="logo-component"
  >
    Logo
  </div>
  <nav
    aria-label="Main"
    class="group/navigation-menu relative flex max-w-max flex-1 items-center justify-center"
    data-orientation="horizontal"
    data-slot="navigation-menu"
    data-viewport="true"
    dir="ltr"
  >
    <div
      style="position: relative;"
    >
      <ul
        class="group flex flex-1 list-none items-center justify-center gap-1"
        data-orientation="horizontal"
        data-slot="navigation-menu-list"
        dir="ltr"
      >
        <li
          class="relative"
          data-slot="navigation-menu-item"
        >
          <a
            data-discover="true"
            href="/"
          >
            <a
              class="data-[active=true]:focus:bg-accent data-[active=true]:hover:bg-accent data-[active=true]:bg-accent/50 data-[active=true]:text-accent-foreground [&_svg:not([class*='text-'])]:text-muted-foreground flex-col gap-1 p-2 [&_svg:not([class*='size-'])]:size-4 group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1"
              data-radix-collection-item=""
              data-slot="navigation-menu-link"
            >
              Home
            </a>
          </a>
        </li>
        <li
          class="relative"
          data-slot="navigation-menu-item"
        >
          <a
            data-discover="true"
            href="/about"
          >
            <a
              class="data-[active=true]:focus:bg-accent data-[active=true]:hover:bg-accent data-[active=true]:bg-accent/50 data-[active=true]:text-accent-foreground [&_svg:not([class*='text-'])]:text-muted-foreground flex-col gap-1 p-2 [&_svg:not([class*='size-'])]:size-4 group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1"
              data-radix-collection-item=""
              data-slot="navigation-menu-link"
            >
              About us
            </a>
          </a>
        </li>
        <li
          class="relative"
          data-slot="navigation-menu-item"
        >
          <a
            data-discover="true"
            href="/contact"
          >
            <a
              class="data-[active=true]:focus:bg-accent data-[active=true]:hover:bg-accent data-[active=true]:bg-accent/50 data-[active=true]:text-accent-foreground [&_svg:not([class*='text-'])]:text-muted-foreground flex-col gap-1 p-2 [&_svg:not([class*='size-'])]:size-4 group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1"
              data-radix-collection-item=""
              data-slot="navigation-menu-link"
            >
              Contact us
            </a>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="absolute top-full left-0 isolate z-50 flex justify-center"
    />
  </nav>
  <div
    class="items-center gap-2 hidden md:flex"
  >
    <svg
      class="lucide lucide-phone size-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
      />
    </svg>
    <span
      class="font-sans font-medium text-foreground"
    >
      (257) 388-6895
    </span>
  </div>
</div>
`;
